website/settings/defaults.py

```diff
@@ -1980,23 +1980,28 @@ class StorageLimits(enum.IntEnum):
     def from_node_usage(cls,  usage_bytes, private_limit=None, public_limit=None):
         """ This should indicate if a node is at or over a certain storage threshold indicating a status. If nodes have
         a custom limit this should indicate that."""
-        custom_limits = dict([(item.name, item.value) for item in cls])
-
-        custom_limits[cls.OVER_PRIVATE.name] = STORAGE_LIMIT_PRIVATE
-        custom_limits[cls.APPROACHING_PRIVATE.name] = STORAGE_LIMIT_PRIVATE * STORAGE_WARNING_THRESHOLD
-
-        custom_limits[cls.OVER_PUBLIC.name] = STORAGE_LIMIT_PUBLIC
-        custom_limits[cls.APPROACHING_PUBLIC.name] = STORAGE_LIMIT_PUBLIC * STORAGE_WARNING_THRESHOLD
+        GBs = 1024 ** 3.0
 
         if public_limit:
-            custom_limits[cls.OVER_PUBLIC.name] = public_limit
-            warning_limit = public_limit * STORAGE_WARNING_THRESHOLD
-            custom_limits['APPROACHING_PUBLIC'] = warning_limit
+            if usage_bytes >= public_limit * GBs:
+                return cls.OVER_PUBLIC
+            elif usage_bytes >= public_limit * STORAGE_WARNING_THRESHOLD * GBs:
+                return cls.APPROACHING_PUBLIC
+
         if private_limit:
-            custom_limits['OVER_PRIVATE'] = private_limit
-            warning_limit = private_limit * STORAGE_WARNING_THRESHOLD
-            custom_limits['APPROACHING_PRIVATE'] = warning_limit
+            if usage_bytes >= private_limit * GBs:
+                return cls.OVER_PRIVATE
+            elif usage_bytes >= private_limit * STORAGE_WARNING_THRESHOLD * GBs:
+                return cls.APPROACHING_PRIVATE
+
+        elif usage_bytes >= STORAGE_LIMIT_PUBLIC * GBs:
+            return cls.OVER_PUBLIC
+        elif usage_bytes >= STORAGE_LIMIT_PUBLIC * STORAGE_WARNING_THRESHOLD * GBs:
+            return cls.APPROACHING_PUBLIC
+        elif usage_bytes >= STORAGE_LIMIT_PRIVATE * GBs:
+            return cls.OVER_PRIVATE
+        elif usage_bytes >= STORAGE_LIMIT_PRIVATE * STORAGE_WARNING_THRESHOLD * GBs:
+            return cls.APPROACHING_PRIVATE
+        else:
+            return cls.DEFAULT
```

Comment:
This approach is still weirdly complicated -- enough so that it accidentally breaks the logic:

If a user has a custom private limit but not a custom public limit, you'll wind up checking the private limit first and fail to notice that they're over the public limit.

More than that, if they have a custom private limit and are under it, you'll skip over the remaining blocks and just return None, causing hard-to-find errors on a future access.

If you pick the "correct" public and private limits at the outset (either the provided value or the default if None), then you can just write one case for each Enum value and be done.